metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -1174,6 +1413,22 @@ def emit_datasource(
             for wu in workunits:
                 self.report.report_workunit(wu)
                 yield wu
+        elif (
```

Comment:
what exactly is happening here? this code is extremely hard to understand. If it's not an embedded data source that has a workbook, then we try to get the luid for the data source's project and emit a container path as the project directly (instead of intermediate workbook) - is this correct?

If yes, please consider a) extracting into more readable, testable methods and b) adding comment that says this!!!!